test/testlib.sh.in

```diff
@@ -370,6 +446,31 @@ cleanup() {
   ) || true
 }
 
+# If we're on a CI server, the test requires superuser privileges to run, and we're not
+# currently a superuser, try running the test as one (if the system configuration allows
+# it — we need passwordless sudo).
+require_root() {
+  if is_root; then
+    return
+  fi
+  if is_ci; then
+    echo "root is required for test $SCRIPTNAME, but we're a regular user; elevating privileges..."
+    if ! command -v sudo 2>/dev/null; then
+      echo >&2 "sudo not found, skipping test $SCRIPTNAME"
+      exit $EXIT_SKIP_TEST
```

Comment:
Should we really be skipping these tests if CI is set? I'd rather have it fail hard if, for some reason, the CI environment is not correctly set up to do passwordless `sudo`.